lib/rubocop/cop/style/combinable_loops.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for places where multiple consecutive loops over the same data
+      # can be combined into a single loop. It is very likely that combining them
+      # will make the code more efficient and more concise.
+      #
+      # It is marked as unsafe, because the first loop can modify
+      # a state that the second loop depends on those two aren't combinable.
```

Comment:
I would tweak slightly => "It is marked as unsafe, because the first loop might modify a state that the second loop depends on; these two aren't combinable."